src/aere_repl.erl

```diff
@@ -141,6 +142,32 @@ apply_command(State, type, I) ->
     TypeStr = aeso_ast_infer_types:pp_type("", Type),
     TypeStrClean = re:replace(TypeStr, ?TYPE_CONTAINER ++ "[0-9]*\\.", "", [global, {return, list}]),
     {aere_msg:output(TypeStrClean), State};
+apply_command(State = #repl_state{contract_state = {_, OldType, _}}, state, I) ->
+    [Expr] = aere_sophia:parse_body(I),
+    Contract = aere_mock:eval_contract([Expr], State),
+    TAst = aere_sophia:typecheck(Contract, [dont_unfold]),
+    {_, Type} = aere_sophia:type_of(TAst, ?USER_INPUT),
+    {StateVal, _, _} = compile_and_run_contract(Contract, State),
+
+    %% No need for the annotations from the mock contract
+    NewType = aeso_syntax:set_ann(aere_mock:ann(), Type),
+    NewValue = aeso_syntax:set_ann(aere_mock:ann(), Expr),
+
+    NewState0 =
+        case OldType =:= NewType of
+            false -> State;
+            true  -> clear_context(State)
+        end,
+    NewState1 = NewState0#repl_state{contract_state = {NewType, NewValue, StateVal}},
+    {_, UsedGas, NewState2} = compile_and_run_contract(Contract, NewState1),
+
+    ResStr = io_lib:format("~p", [StateVal]),
```

Comment:
This becomes a pattern. Printing a value with optional gas should go to a separate function

Reply:
Done here 6a33514c9bdafb364b6aba00440e6f39f3928202